redux/react/ConnectedComponentMacro.hx

```diff
@@ -0,0 +1,665 @@
+				}
+
+				return field;
+			}
+		}
+
+		return null;
+	}
+
+	static function addMapDispatchToPropsWarnings(connectedFunctions:Array<Field>)
+	{
+		for (f in connectedFunctions)
+			Context.warning(
+				'@:connect functions will not be connected by the macro '
+				+ 'when you define mapDispatchToProps',
+				f.pos
+			);
+	}
+
+	static function addMapDispatchToProps(
+		fields:Array<Field>,
+		connectedFunctions:Array<Field>,
+		propsType:ComplexType,
+		componentPropsType:ComplexType,
+		inClass:ClassType
+	) {
+		fields.push({
+			name: 'mapDispatchToProps',
+			doc: null,
+			meta: [],
+			access: [AStatic],
+			kind: FFun({
+				args: [
+					{type: macro :redux.Redux.Dispatch, name: '__dispatch'},
+					{type: propsType, name: 'ownProps', opt: true}
+				],
+				params: [],
+				ret: getPartialType(componentPropsType),
+				expr: mapDispatchToPropsExpr(connectedFunctions, inClass)
+			}),
+			pos: inClass.pos
+		});
+	}
+
+	static function mapDispatchToPropsExpr(connectedFunctions:Array<Field>, inClass:ClassType)
+	{
+		var propsFields = [for (connectedFunction in connectedFunctions)
+			{
+				field: connectedFunction.name,
+				expr: macro $i{connectedFunction.name}.bind(__dispatch)
+			}
+		];
+
+		return {
+			expr: EBlock([{
+				expr: EReturn({
+					expr: EObjectDecl(propsFields),
+					pos: inClass.pos
+				}),
+				pos: inClass.pos
+			}]),
+			pos: inClass.pos
+		};
+	}
+
+	static function addConnected(fields:Array<Field>, inClass:ClassType)
+	{
+		fields.push({
+			name: 'Connected',
+			access: [AStatic, APublic],
+			kind: FProp('get', 'null', macro :react.React.CreateElementType),
+			pos: inClass.pos
+		});
+
+		fields.push({
+			name: 'get_Connected',
+			doc: null,
+			meta: [],
+			access: [AStatic],
+			kind: FFun({
+				args: [],
+				params: [],
+				ret: macro :react.React.CreateElementType,
+				expr: getConnectedExpr()
+			}),
+			pos: inClass.pos
+		});
+	}
+
+	static function getConnectedExpr()
+	{
+		return macro {
+			if (Connected == null)
+				Connected = redux.react.ReactRedux.connect(
+					mapStateToProps,
+					mapDispatchToProps,
+					mergeProps,
+					options
+				)(render);
+
+			return Connected;
+		};
+	}
+
+	static function addRender(
+		fields:Array<Field>,
+		componentType:Type,
+		componentPropsType:ComplexType,
+		inClass:ClassType
+	) {
+		var componentName = switch (componentType) {
+			case TInst(compClass, _):
+				compClass.get().name;
+
+			default:
+				Context.fatalError(
+					'Invalid wrapped component for React Container',
+					inClass.pos
+				);
+		};
+
+		fields.push({
+			name: 'render',
+			doc: null,
+			meta: [],
+			access: [AStatic],
+			kind: FFun({
+				args: [{
+					type: componentPropsType,
+					name: 'props'
+				}],
+				params: [],
+				ret: null,
+				expr: renderExpr(componentName, inClass.pos)
+			}),
+			pos: inClass.pos
+		});
+	}
+
+	static function renderExpr(componentName:String, pos:Position)
+	{
+		var jsxStr = '<$' + componentName + ' {...props} />';
```

Comment:
`$` is not needed  - that's really only a convenience for "click to navigate" in editors.
I had some doubts but I think you can pass children this way as well.